warehouse/models/staging/gtfs_guidelines/stg_gtfs_guidelines__no_expired_services.sql

```diff
@@ -0,0 +1,72 @@
+WITH feed_guideline_index AS (
+    SELECT * FROM {{ ref('stg_gtfs_guidelines__feed_guideline_index') }}
+    WHERE check = {{ no_expired_services() }}
+),
+
+stalest_calendar_services AS (
+   SELECT
+       calitp_itp_id,
+       calitp_url_number,
+       calitp_extracted_at,
+       calitp_deleted_at,
+       MIN(end_date) AS end_date
+    FROM {{ ref('calendar_clean') }}
+   GROUP BY 1, 2, 3, 4
+),
+
+stalest_calendar_dates_services AS (
+   SELECT
+       calitp_itp_id,
+       calitp_url_number,
+       calitp_extracted_at,
+       calitp_deleted_at,
+       MIN(date) AS date
```

Comment:
This one may be more complicated actually.... I think you may need to group by service ID, get the max by service ID, and then get the min over those maxes? Because I think as written it will just get the first date with service defined, and it is ok if that is in the past?

Reply:
Would be fine if you want to just merge as it was (calendar only) and then iterate more on this, didn't mean to scope creep you 😅

Reply:
It's all good - doesn't hurt to get it right while it's top of mind.

Reply:
Thanks for thinking this through, @lauriemerrell. Yes, as always you are right. I under-thought about how calendar_dates works, and so my simple rule was "if a date in the past appears, it's bad". Your way of thinking of it reflects the guideline much better.

Reply:
I updated this, although the more I think about it I think it could go either way. Going to keep these changes at least for now, though.

Reply:
I think that the old way would give you a ton of false failures, because they would mean that any feed using `calendar_dates` for service would get marked as expired because the first date is almost always going to be in the past. 

Agree that this implementation may still be buggy.... For example, for feeds that use *both* `calendar` and `calendar_dates`, this may cause them to fail the `calendar_dates` check for example if a holiday exception is defined for Labor Day on a service that's in effect from August to December. That service is not expired in `calendar` (ends in December) but is in `calendar_dates` (because only Labor Day in September will be listed). 

So I think getting max date by service ID across both may be even more robust (i.e., outer join your two checks and get the max date that appears in either table by service ID )? But defer to you on how far to go with this; I just suspect you may see still unduly high failure rates. I'll approve and defer to you.